server/valgrind-out.txt

```diff
@@ -1,113 +1,129 @@
+--3955-- REDIR: 0x493d860 (libc.so.6:wcscpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4920820 (libc.so.6:strpbrk) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4920100 (libc.so.6:index) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4920630 (libc.so.6:strlen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4929bb0 (libc.so.6:memrchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4921850 (libc.so.6:strcasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x49213d0 (libc.so.6:memchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x493d930 (libc.so.6:wcslen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4920ae0 (libc.so.6:strspn) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4921750 (libc.so.6:stpncpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x49216f0 (libc.so.6:stpcpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4922c90 (libc.so.6:strchrnul) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x49218a0 (libc.so.6:strncasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--3955-- REDIR: 0x4a09500 (libc.so.6:__strrchr_avx2) redirected to 0x483ea10 (rindex)
+--3955-- REDIR: 0x4a0cb60 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x48428e0 (memset)
+--3955-- REDIR: 0x4a096d0 (libc.so.6:__strlen_avx2) redirected to 0x483ef40 (strlen)
+--3955-- REDIR: 0x491b0e0 (libc.so.6:malloc) redirected to 0x483b780 (malloc)
+--3955-- REDIR: 0x4a0c6c0 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4843660 (mempcpy)
+--3955-- REDIR: 0x4a09310 (libc.so.6:__strchrnul_avx2) redirected to 0x4843540 (strchrnul)
+--3955-- REDIR: 0x4a0ac10 (libc.so.6:__strcpy_avx2) redirected to 0x483f090 (strcpy)
+--3955-- REDIR: 0x491cb10 (libc.so.6:calloc) redirected to 0x483dce0 (calloc)
+--3955-- REDIR: 0x491b6d0 (libc.so.6:free) redirected to 0x483c9d0 (free)
+--3955-- REDIR: 0x4a0c6e0 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x48429f0 (memmove)
+--3955-- REDIR: 0x4920fa0 (libc.so.6:__GI_strstr) redirected to 0x4843ca0 (__strstr_sse2)
+--3955-- REDIR: 0x4a05cc0 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x48421e0 (bcmp)
+--3955-- REDIR: 0x4a04bd0 (libc.so.6:__strcmp_avx2) redirected to 0x483fed0 (strcmp)
+--3955-- REDIR: 0x491be80 (libc.so.6:realloc) redirected to 0x483df30 (realloc)
+==3955== 
+==3955== HEAP SUMMARY:
+==3955==     in use at exit: 80 bytes in 1 blocks
+==3955==   total heap usage: 958 allocs, 957 frees, 339,926 bytes allocated
+==3955== 
+==3955== Searching for pointers to 1 not-freed blocks
+==3955== Checked 91,888 bytes
+==3955== 
+==3955== 80 bytes in 1 blocks are still reachable in loss record 1 of 1
+==3955==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==3955==    by 0x109DAE: main (aesdsocket.c:293)
```

Comment:
@TejaswiniLakshminarayan this is pointing to a problem with your design, see comments above.